src/main/java/org/ow2/petals/deployer/utils/RuntimeModelDeployer.java

```diff
@@ -0,0 +1,177 @@
+
+        LOG.fine("Deploying model (" + serviceUnits.size() + " service units)");
+
+        String hostname = container.getHostname();
+        int port = container.getPort();
+        String user = container.getUser();
+        String password = container.getPassword();
+
+        petalsAdmin.connect(hostname, port, user, password);
+
+        for (RuntimeServiceUnit serviceUnit : serviceUnits) {
+            deployRuntimeServiceUnit(serviceUnit, model);
+        }
+
+        LOG.fine("Model deployed");
+    }
+
+    private void deployRuntimeServiceUnit(RuntimeServiceUnit serviceUnit, RuntimeModel model)
+            throws IOException, JBIDescriptorException, ArtifactStartedException, ArtifactNotDeployedException,
+            ArtifactNotFoundException, ArtifactAdministrationException, RuntimeModelDeployerException {
+        String suId = serviceUnit.getId();
+        File suFile = Files.createTempFile(suId, ".zip").toFile();
+        FileUtils.copyURLToFile(serviceUnit.getUrl(), suFile, ModelDeployer.CONNECTION_TIMEOUT,
+                ModelDeployer.READ_TIMEOUT);
+        Jbi jbi = jdb.buildJavaJBIDescriptorFromArchive(suFile);
+        ServiceAssembly jbiSa = jbi.getServiceAssembly();
+
+        if (!deployedServiceUnits.contains(suId)) {
+            LOG.fine("Deploying service assembly");
+            for (org.ow2.petals.jbi.descriptor.original.generated.ServiceUnit jbiSu : jbiSa.getServiceUnit()) {
+                String compName = jbiSu.getTarget().getComponentName();
+                if (!deployedComponents.contains(compName)) {
+                    RuntimeComponent component = model.getContainers().iterator().next().getComponent(compName);
+                    if (component == null) {
+                        throw new RuntimeModelDeployerException("Component " + compName + " (needed by service unit "
+                                + jbiSu.getIdentification().getName() + ") not found");
+                    }
+                    deployRuntimeComponent(component);
+                    deployedComponents.add(compName);
+                }
+            }
+            ServiceAssemblyLifecycle saLifecycle = artifactLifecycleFactory.createServiceAssemblyLifecycle(
+                    new org.ow2.petals.admin.api.artifact.ServiceAssembly(jbiSa.getIdentification().getName()));
+            saLifecycle.deploy(suFile.toURI().toURL());
+            saLifecycle.start();
+            for (org.ow2.petals.jbi.descriptor.original.generated.ServiceUnit jbiSu : jbiSa.getServiceUnit()) {
+                String suName = jbiSu.getIdentification().getName();
+                if (!deployedServiceUnits.contains(suName)) {
+                    deployedServiceUnits.add(suName);
+                    LOG.fine("Service unit " + suName + " deployed and started");
+                } else {
+                    throw new RuntimeModelDeployerException("Service unit " + suName + " already deployed");
```

Comment:
We should be able to apply a same model over and over again without having errors.
If it is already deployed, then it is not important. I would rather log the information, instead of throwing an exception.

Reply:
Je suis pas completement d'accord. Il faut les deux modes:
* de quoi faire un déploiement qui t'indique les erreurs
* et un qui sait quoi faire en fonction des erreurs, c'est ce que j'appelerai le mode 'forcé' (si vous avez un meilleur terme, je suis preneur)
Le comprtement est fonction d'un paramètre de la part du client de la librairie petals-deployer

Reply:
See #14